tls/s2n_x509_validator.c

```diff
@@ -399,6 +400,66 @@ S2N_RESULT s2n_x509_validator_read_asn1_cert(struct s2n_stuffer *cert_chain_in_s
     return S2N_RESULT_OK;
 }
 
+/**
+* Validates that each certificate in a peer's cert chain contains only signature algorithms in a security policy's
+* certificate_signatures_preference list.
+*/
+S2N_RESULT s2n_validator_check_cert_preferences(struct s2n_connection *conn, X509 *cert)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(cert);
+
+    const struct s2n_security_policy *security_policy;
```

Comment:
nit
```suggestion
    const struct s2n_security_policy *security_policy = NULL;
```

Reply:
clang-tidy can check this for us. #4404 will fix all current violations, and then I'll add a check in CI to alert on this.